src/editor/tools/multi_select.h

```diff
@@ -52,17 +52,12 @@ struct MultiSelect {
 	}
 
 	[[nodiscard]] int32_t get_random_enabled() const {
-		int32_t rand_value = static_cast<int32_t>(static_cast<double>(get_nr_enabled()) *
-		                                          RNG::static_rand() / (RAND_MAX + 1.0));
-
-		for (int32_t item : enabled_) {
-			if (rand_value == 0) {
-				return item;
-			}
-			--rand_value;
+		if (get_nr_enabled() < 1) {
+			return -1;
 		}
-
-		return -1;
+		auto it = enabled_.begin();
+		std::advance(it, RNG::static_rand(get_nr_enabled()));
```

Comment:
as the result of static_rand is exclusive of the number of elements, I am not sure if the last element of all enabled elements will ever be picked.

Reply:
Yes because indices are zero-based. If for example there are 5 elements, they are indexed 0,1,2,3,4. `static_rand` returns `(random number) % 5` which generates the numbers 0,1,2,3,4.